File: src/main/java/dev/hoot/api/packets/MovementPackets.java
package dev.hoot.api.packets;

import dev.hoot.api.game.Game;
import net.runelite.api.Client;
import net.runelite.api.coords.WorldPoint;
import net.runelite.api.packets.ClientPacket;
import net.runelite.api.packets.PacketBufferNode;

public class MovementPackets
{
	public static void sendMovement(int worldX, int worldY)
	{
		MovementPackets.sendMovement(worldX, worldY, false);
	}

	public static void sendMovement(WorldPoint worldPoint, boolean ctrlDown)
	{
		MovementPackets.sendMovement(worldPoint.getX(), worldPoint.getY(), ctrlDown);
	}

	public static void sendMovement(WorldPoint worldPoint)
	{
		MovementPackets.sendMovement(worldPoint, false);
	}

	public static void sendMovement(int worldPointX, int worldPointY, boolean ctrlDown)
	{
		createMovement(worldPointX, worldPointY, ctrlDown).send();
	}

	public static PacketBufferNode createMovement(int worldPointX, int worldPointY, boolean ctrlDown)
	{
		var client = Game.getClient();
		var clientPacket = Game.getClientPacket();
		var packetBufferNode = Game.getClient().preparePacket(clientPacket.MOVE_GAMECLICK(), client.getPacketWriter().getIsaacCipher());
		packetBufferNode.getPacketBuffer().writeByte(5);
		packetBufferNode.getPacketBuffer().writeByte(ctrlDown ? 2 : 0);	// Old: packetBufferNode.getPacketBuffer().writeByteSub(ctrlDown ? 2 : 0);
		packetBufferNode.getPacketBuffer().writeShort(worldPointX);	// Old: packetBufferNode.getPacketBuffer().writeShortAddLE(worldPointY);
		packetBufferNode.getPacketBuffer().writeShortLE(worldPointY);	// Old: packetBufferNode.getPacketBuffer().writeShort(worldPointX);
		return packetBufferNode;
	}
}

File: src/main/kotlin/meteor/api/packets/MovementPackets.kt
package meteor.api.packets

import net.runelite.api.Client
import net.runelite.api.coords.WorldPoint

object MovementPackets {
    fun sendMovement(worldX: Int, worldY: Int) {
        sendMovement(worldX, worldY, false)
    }

    fun sendMovement(worldPoint: WorldPoint, ctrlDown: Boolean) {
        sendMovement(worldPoint.x, worldPoint.y, ctrlDown)
    }

    fun sendMovement(worldPoint: WorldPoint) {
        sendMovement(worldPoint, false)
    }

    fun sendMovement(worldPointX: Int, worldPointY: Int, ctrlDown: Boolean) {
        val client: Client = Main.client
        val clientPacket = Main.clientPacket
        val packetBufferNode =
            client.preparePacket(clientPacket.MOVE_GAMECLICK(), client.packetWriter.isaacCipher)
        packetBufferNode.packetBuffer.writeByte(5)
        packetBufferNode.packetBuffer.writeByteAdd(if (ctrlDown) 2 else 0)
        packetBufferNode.packetBuffer.writeShortLE(worldPointX)
        packetBufferNode.packetBuffer.writeShortLE(worldPointY)
        client.packetWriter.queuePacket(packetBufferNode)
    }
}